### PR TITLE
Array padding rule

### LINKED ISSUE #1045: Require whitespace inside inline array literals
good

``` ts
const x = [ "one", "two" ];
```

bad

``` ts
const x = ["one", "two"];
```

### PR DESCRIPTION
**Pull request by leeavital**
@JKillian has generously offered to finish this 😄 

---

edit by @adidahiya: fixes #1045

### CODE REVIEW

File: test/rules/arrayPadding/test.ts.lint [hunk 1]
```diff
@@ -0,0 +1,7 @@
+["abc"]
+~~~~~~~ [array padding]
+
+["this", "is", "ok"]
```
Comment (thasner) [inline test/rules/arrayPadding/test.ts.lint]: shouldn't this break the rule?

**Comment by adidahiya** [comment]
I think this should be called `array-bracket-spacing` like ESLint: http://eslint.org/docs/rules/array-bracket-spacing.

Related: https://github.com/buzinas/tslint-eslint-rules/issues/57

**Comment by thasner** [comment]
sgtm

**Comment by JKillian** [comment]
I'm fine with `array-bracket-spacing`, although we could also make one rule that has options for object-brace-spacing, import-brace-spacing, jsx-attribute-brace-spacing and this. Or we can just keep them all as separate rules.

**Comment by chrismbarr** [comment]
Reporting a failure message that just says
> array padding

Isn't very descriptive or helpful. Perhaps something that actually describes what the issue is and how to fix it, like:

>Arrays require whitespace between brackets

Also, does this rule consider initializing arrays: ` arr = [];`
or would you have to initialize like `arr = [ ];` ?

File: test/rules/arrayPadding/test.ts.lint (hunk 1 above)
Comment (chrismbarr) [inline test/rules/arrayPadding/test.ts.lint]: Ageed.  I don't see the difference.  I would expect:
`[ "this", "is", "ok" ]`
not
`["this", "is", "ok"]`

**Comment by adidahiya** [comment]
👍  for more descriptive error message.

And I think `arr = []` should still be allowed; enforcing `arr = [ ]` feels unnecessary.

**Comment by JKillian** [comment]
Sorry, this is very much not complete. I had told @leeavital he could just throw whatever he had up here and I would finish it, but very clearly I haven't done that yet...

**Comment by adidahiya** [comment]
this PR is old, open a new one with rebased commits if you'd like to take up this task

**Pull request closed without merging**